### ISSUE TITLE
joint training hyper parameters

### ISSUE THREAD

**Issue by lxa9867** [wjn922/ReferFormer#7]
Hi,

Thank you for sharing your work. I am writing to inquire the hyper parameters used for joint training. The arxiv paper mentioned that 
![image](https://user-images.githubusercontent.com/27429220/161579861-5045193c-3d7d-478f-b7d9-88ec4e92e44d.png)
which says that the joint training uses 32 V100 GPUs and 2 video clips for each GPU. 

I consider it means 32G V100 GPU. But I think it's not possible to add 2 video clip within 32G memory. I cannot reproduce the result using 8 V100 32G GPU with 1 clip per GPU, would you like to give me some advice? Thank you!

**Comment by lxa9867** [comment]
BTW, would you like to share your script for pretraining, thanks!

**Comment by wjn922** [comment]
Hi, 

We use 32 V100 GPUs with 32G memory. And sorry for the mistake, it is actually is 1 for joint training and 2 for pretraining.
The joint training and pretraining stages would take around 1~2 days depends on the backbones.

We do not support multi-node training in the repo now. For single node training, the pretraining script is 
```python
python3 -m torch.distributed.launch --nproc_per_node=8  --use_env \
        main_pretrain.py  --dataset_file all --binary --with_box_refine \
                                      --batch_size 2 --num_frames 1 \
                                     --epochs 12 --lr_drop 8 10 \
                                     [backbone]
```

**Comment by lxa9867** [comment]
Thank you for your information!

**Comment by lxa9867** [comment]
Did you freeze text encoder during pretraining?

**Comment by wjn922** [comment]
The text encoder is not frozen for pretraining. While for joint training, it is frozen.

**Comment by lxa9867** [comment]
I think I found the reason I cannot reproduce the result...The paper has some ambiguity for this part. Thank you.

btw, there is a bug in the main_pretrain.py, the lr scheduler should be multiStep instead of Step, the current code cannot correctly running with your given cmd. I leave the comments here to help others save time.

**Comment by lxa9867** [comment]
Hi, 

May I ask why you freeze the text encoder during joint training? Since the joint training doesn't need a pretraining process, I would assume the text encoder is not trained? Thank you!

**Comment by wjn922** [comment]
We have experimented whether to freeze the text encoder with the R50 backbone. And we found freezing the text encoder would have 1+ points gain.

**Comment by lxa9867** [comment]
Thank you very much!
